repos/spack_repo/builtin/packages/py_xhistogram/package.py

```diff
@@ -18,6 +18,16 @@ class PyXhistogram(PythonPackage):
     version("0.3.2", sha256="56b0751e1469eaed81710f644c8ba5c574b51883baa2feee26a95f2f708f91a1")
 
     depends_on("py-setuptools", type="build")
+    depends_on("py-versioneer", type="build")
+    depends_on("py-versioneer@0.29:", type="build", when="^python@3.12:")
+
     depends_on("py-xarray@0.12:", type=("build", "run"))
     depends_on("py-dask@2.3:+array", type=("build", "run"))
     depends_on("py-numpy@1.17:", type=("build", "run"))
+
+    # compatibility with python 3.12
+    patch(
+        "https://github.com/xgcm/xhistogram/commit/6eef6c697d95ea70883a5ff6ee2f3e7188eaa4c5.patch?full_index=1",
+        sha256="2a508aded3ac57dfe7c87ff3e27124400daf04391759009fea209ce5cb0067fe",
+        when="^python@3.12",
```

Comment:
We should avoid commits on forks. We should probably version guard patches so they don't break on the next package upgrade. Does this only apply to python 3.12, or also higher?

Reply:
What is best approach on this? standalone patch file? It is `3.12:` afaik -- good point!

Reply:
Depends. What's the timescale when this might get merged?

Reply:
In that case I think a local copy seems appropriate (it's small anyway).